src/common/types/data_chunk.cpp

```diff
@@ -532,12 +535,18 @@ void SetArrowChild(DuckDBArrowArrayChildHolder &child_holder, const LogicalType
 		//! step 1: figure out total string length:
 		idx_t total_string_length = 0;
 		auto string_t_ptr = FlatVector::GetData<string_t>(*child_holder.vector);
+		auto uint64_t_ptr = FlatVector::GetData<uint64_t>(*child_holder.vector);
```

Comment:
Can we turn this conversion into a templated function that takes either a `ArrowVarcharConversion` or a `ArrowUUIDConversion` struct, with various static members that enable that specific conversion, e.g. something like:

```cpp
struct ArrowUUIDConversion {
    using internal_type_t = uint64_t;

    static idx_t GetStringLength(uint64_t value) {
         return 36;
    }
    ...
};

struct ArrowVarcharConversion {
    using internal_type_t = string_t;

    static idx_t GetStringLength(string_t value) {
         return value.GetSize();
    }
    ...
};
```